cupy/manipulation/add_remove.py

```diff
@@ -12,8 +13,62 @@
 
 # TODO(okuta): Implement resize
 
+_first_nonzero_krnl = core.ReductionKernel(
+    'raw S data, T indices, T len',
+    'T y',
+    'data[_j] ? _j : len',
```

Comment:
How about comparing with zero explicitly? It will work also for complex inputs.
```suggestion
    'data[_j] == S(0) ? _j : len',
```

Reply:
Anyway, I feel `T` is preferred for the type name.